### PR TITLE
Updated the Promises, a debug mode is introduced, added new payment link generate function and fixed other endpoint issues. 

### PR DESCRIPTION
**Pull request by chamathpali**
Updated the Promises, a debug mode is introduced, added new payment link generate function and fixed other endpoint issues. 
Added readme and integration samples.

### CODE REVIEW

**Comment by AshanFernando** [comment]
@chamathpali Remove node_modules folder from the commit

**Comment by chamathpali** [comment]
Okay I'll update

File: index.js [hunk 1]
```diff
@@ -3,59 +3,120 @@ var http = require('http'),
...
+
+// Adding new Subscription 
+exports.addNewSubscription = function(product_id, key, body) {
+    if (debug) {
+        return new BbPromise(function(resolve, reject) {
+            resolve('{ "status":"success", "message":"Subscription created without notifications and without invoice" }')
+        });
+    } else {
+
+        var url = 'https://app.earnshark.com/prod/product/' + product_id + '/addsubscriptionfromapi?key=' + key;
```
Comment (AshanFernando) [inline index.js]: Define the url outside of code

File: index.js [hunk 2]
```diff
@@ -3,59 +3,120 @@ var http = require('http'),
...
+            options = {
+                url: url,
+                method: 'GET',
+                headers: headers
+            };
+
+        return new BbPromise(function(resolve, reject) {
+            request(options, function(error, response, body) {
+                if (error) {
+                    //  console.log(error);
```
Comment (AshanFernando) [inline index.js]: Remove commented code

File: index.js [hunk 3]
```diff
@@ -3,59 +3,120 @@ var http = require('http'),
...
         });
-    }).on('error', (e) => {
-        console.error(e);
-    });
+    }
 };
+
+// Generate the payment url for license payments
+exports.getPaymentURL  = function(product_id,key,account_id,redirect){
+    return 'http://earnsharkbeta.com.s3-website-eu-west-1.amazonaws.com/payment.html?redirect='+redirect+'&productID='+product_id+'&accountID='+account_id+'&key='+key;
```
Comment (AshanFernando) [inline index.js]: Define the url outside the method

**Pull request merged**